angular-ngrx-scss/src/app/state/user/user.effects.ts

```diff
@@ -12,16 +21,38 @@ import {
   fetchUserTopReposError,
   fetchUserTopReposSuccess,
 } from './user.actions';
+import { UserState } from './user.state';
 
 @Injectable()
 export class UserEffects {
   loadUser$ = createEffect(() => {
     return this.actions$.pipe(
       ofType(fetchUserData),
-      switchMap(() =>
-        this.userService.getAuthenticatedUserInfo().pipe(
-          map((data) => fetchUserDataSuccess({ userData: data })),
-          catchError((error) => of(fetchUserDataError({ error }))),
+      switchMap(({ username }) =>
+        this.store.select(selectAuthUserName).pipe(
+          skipWhile((name) => name === ''),
+          mergeMap(() =>
+            this.userService.getUserInfo(username).pipe(
+              map((userData) => {
+                const user: UserState = {
+                  avatar: userData.avatar_url,
+                  bio: userData.bio,
+                  blog: userData.blog,
+                  company: userData.company,
+                  email: userData.email,
+                  followers: userData.followers,
+                  following: userData.following,
+                  location: userData.location,
+                  name: userData.name,
+                  twitter_username: userData.twitter_username,
+                  username: userData.login,
+                  type: userData.type,
+                };
+                return fetchUserDataSuccess({ userData: user });
```

Comment:
Sorry Chris, I'm not sure I understand. Do you mean we should create some functions in the state files that would handle mapping from API response to app response, and then pull in those functions and use them in the effects? Instead of directly doing the mapping here?